### PR TITLE
Section Create/Destroy API route

### PR DESCRIPTION
**Pull request by mimischly7**
## Motivation and Context

Currently, most models are associated with two different controllers, one "main" controller responsible for "business logic", and one api-controller mostly responsible for CRUD operations (on that model). The `Section` model does not yet adhere to this pattern, hence this pull request, which includes a new `api/sections_controller` controller, featuring a new api route responsible for creating and deleting a section.

## Your Changes

**Description**:
- Created a new `api/sections_controller` with a `destroy` and 'create' action.
- Added 'resources :sections ...' line to the `~/config/routes.rb` file (in the appropriate location so that the new api URI's have the appropriate nested format, i.e. .../courses/.../sections/...).

**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)

## Testing

Created a new rspec file (`api/sections_controller_spec.rb`) to test the functionality of this new API route, for authorized and unauthorized attempts.

## Checklist
- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [x] I have added tests for my changes. 
- [x] I have updated the Changelog.md file. 
- [x] Link for PR regarding additional API documentation: https://github.com/MarkUsProject/Wiki/pull/200

### CODE REVIEW

File: spec/controllers/api/sections_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      end
+    end
+
+    context 'Delete' do
+      it 'successfully deletes section' do
+        delete :destroy, params: { course_id: course.id, id: section }
+        expect(response).to have_http_status(:ok)
+        # The below two check the same thing, not sure which is preferable
+        expect { course.sections.find(section.id) }.to raise_error(ActiveRecord::RecordNotFound)
+        expect(course.sections.exists?(section.id)).to be_falsey
```
Comment (david-yz-liu) [inline spec/controllers/api/sections_controller_spec.rb:42]: this is the one that's preferred

**Review by david-yz-liu — COMMENTED** [review]
@mimischly7 @Bruce-8 good work. I left one inline comment. In addition to that, the Coveralls report suggests the need for an additional test case for the `#delete` route.

**Review by david-yz-liu — COMMENTED** [review]
@mimischly7 @Bruce-8 sorry one more thing, can you please make a PR to update the Wiki API page: https://github.com/MarkUsProject/Wiki/blob/master/RESTful-API.md

**Comment by mimischly7** [comment]
> @mimischly7 @Bruce-8 sorry one more thing, can you please make a PR to update the Wiki API page: https://github.com/MarkUsProject/Wiki/blob/master/RESTful-API.md

https://github.com/MarkUsProject/Wiki/pull/200

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**